src/goto-programs/interpreter.cpp

```diff
@@ -359,7 +366,7 @@ void interpretert::step()
   case CATCH:
     break;
   default:
-    throw "encountered instruction with undefined instruction type";
+    UNREACHABLE; // I'm assuming we'd catch such a thing before this point
```

Comment:
Sounds like you're saying it is a consequence of the goto program being well-formed ;-)